server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -833,26 +834,51 @@ private void construct(
         List<? extends SlowLogFieldProvider> slowLogFieldProviders = pluginsService.loadServiceProviders(SlowLogFieldProvider.class);
         // NOTE: the response of index/search slow log fields below must be calculated dynamically on every call
         // because the responses may change dynamically at runtime
-        SlowLogFieldProvider slowLogFieldProvider = indexSettings -> {
-            final List<SlowLogFields> fields = new ArrayList<>();
-            for (var provider : slowLogFieldProviders) {
-                fields.add(provider.create(indexSettings));
-            }
-            return new SlowLogFields() {
-                @Override
-                public Map<String, String> indexFields() {
-                    return fields.stream()
-                        .flatMap(f -> f.indexFields().entrySet().stream())
-                        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        SlowLogFieldProvider slowLogFieldProvider = new SlowLogFieldProvider() {
```

Comment:
The point of this is that it will merge all field providers provided by the plugins into one (in practice I think that's only the one in the security plugin). So this should ideally have an implementation for merging `queryFields` too and then this instance should be passed to the `PluginService` (instead of the list of providers). 

If you don't want to implement the merge for `indexFields` and `searchFields` in the provider without index settings you could add an empty default implementation in the interface in the same way you did for `queryFields`.